rt/vm/longs/src/main/scala/org/apidesign/bck2brwsr/vm4brwsr/Module.scala

```diff
@@ -0,0 +1,46 @@
+/**
+ * Back 2 Browser Bytecode Translator
+ * Copyright (C) 2012-2017 Jaroslav Tulach <jaroslav.tulach@apidesign.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. Look for COPYING file in the top folder.
+ * If not, see http://opensource.org/licenses/GPL-2.0.
+ */
+package org.apidesign.bck2brwr.vm4brwsr
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.scalajs.runtime.RuntimeLong
+
+@JSExportAll
+@JSExportTopLevel( "Longs" )
+object Module {
+  def to64(lo: Int, hi: Int): Any = new RuntimeLong(lo, hi)
+  def high32(a: RuntimeLong): Int = a.hi;
+  def low32(a: RuntimeLong): Int = a.lo;
```

Comment:
In order not to rely on the API of `RuntimeLong` at the Scala.js level (which is not really a *public* API), you can write those three methods as:
```scala
def to64(lo: Int, hi: Int): Long = (lo.toLong & 0xffffffffL) | (hi.toLong << 32)
def high32(a: Long): Int = (a >>> 32).toInt
def low32(a: Long): Int = a.toInt
```
The Scala.js optimizer is good enough to optimize away all of that encoding, so those are compiled down to:
```js
$c_Lhelloworld_HelloWorld$.prototype.to64__I__I__J = (function(lo, hi) {
  return new $c_sjsr_RuntimeLong(lo, hi)
});
$c_Lhelloworld_HelloWorld$.prototype.high32__J__I = (function(a) {
  var lo = a.hi$2;
  return lo
});
$c_Lhelloworld_HelloWorld$.prototype.$$js$exported$meth$low32__J__O = (function(a) {
  return a.lo$2
});
```

Reply:
The `longs` project isn't part of the build, so I guess this is not that urgent. I know that `RuntimeLong` is an implementation detail; should it ever disappear, I do the bit manipulation as you suggest.